apps/wallet/src/ui/app/pages/home/transfer-coin/PreviewTransfer.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useMemo } from 'react';
+
+import { Text } from '_app/shared/text';
+import { IconTooltip } from '_app/shared/tooltip';
+import { TxnAddress } from '_components/receipt-card/TxnAddress';
+import { TxnAmount } from '_components/receipt-card/TxnAmount';
+import { parseAmount } from '_helpers';
+import { useAppSelector, useFormatCoin, useCoinDecimals } from '_hooks';
+import { GAS_SYMBOL, GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
+
+export type TransferCoinFormProps = {
+    coinType: string;
+    gasCostEstimation: number | null;
+    to: string;
+    amount: string;
+};
+
+export function PreviewTransfer({
+    coinType,
+    gasCostEstimation,
+    to,
+    amount,
+}: TransferCoinFormProps) {
+    const accountAddress = useAppSelector(({ account }) => account.address);
+    const [decimals] = useCoinDecimals(coinType);
+    const amountWithoutDecimals = useMemo(
+        () => parseAmount(amount, decimals),
+        [amount, decimals]
+    );
+
+    const [formattedGas] = useFormatCoin(gasCostEstimation, GAS_TYPE_ARG);
+
+    return (
+        <div className="divide-y divide-solid divide-steel/20 divide-x-0 flex flex-col px-2.5 w-full">
+            <TxnAmount
+                amount={amountWithoutDecimals.toString()}
```

Comment:
How are we handling the isPaySuiAll case? do we want to add the ~ here as well?

Reply:
done